rfd/0098-registered-openssh-nodes.md

```diff
@@ -0,0 +1,111 @@
+
+#### Manual certificate rotation
+
+Users not using EC2 discovery mode and hosting registered OpenSSH nodes themselves will have to rotate the OpenSSH CA manually and generate new SSH host certificates. They can do so by:
+
+1. Exporting the OpenSSH CA: `tctl auth export --type=openssh | sed "s/cert-authority\ //"`
+2. Write the exported CA to the configuration folder of the node's SSH daemon, ex. `/etc/ssh/` for OpenSSH on Linux
+3. Replace the `TrustedUserCAKeys` setting in `sshd_config` to point to the newly created CA
+4. Generate a new host certificate: `tctl auth sign --host=<addr,hostname> --format=openssh --out=host-cert`
+5. Replace the existing host certificate with the new one on the node and set `HostKey` and `HostCertificate` in `sshd_config`
+
+### RBAC
+
+When a user sends a request to a Proxy to connect to a node, the Proxy will attempt to find a node resource by either its hostname or IP, whichever the user specified. If the resource exists and has the `openssh` `sub_kind`, an RBAC check will be performed. If the resource does not exist or isn't an `openssh` node, the connection flow will continue as normal.
+
+### Security
+
+Both RBAC checks and session recording for registered OpenSSH nodes require that users connect through a Proxy. If users are able to connect to registered OpenSSH nodes directly, they can bypass both features. Currently the User CA public key is copied to OpenSSH nodes and configured to be trusted by `sshd`. This means OpenSSH nodes will accept certificates that are signed by the User CA. The problem with that is when Teleport users authenticate with a cluster, the Auth server replies with a certificate that is signed by the User CA. Users could potentially use this certificate to directly connect to registered OpenSSH nodes.
+
+Using the new OpenSSH CA to sign certificates used to authenticate with registered OpenSSH nodes solves this problem, as Teleport users do not have access to any certificates signed by the OpenSSH CA.
+
+
+#### Forwarding SSH Agents
+
+If a certificate signed by the OpenSSH CA is added to a forwarded SSH agent, then anyone with access to the forwarded SSH agent socket will have access to any other registered OpenSSH node that trusts OpenSSH CA. To prevent this, certificates signed by the OpenSSH CA will never be added to SSH agents. Futhermore, forwarding SSH agents will not be required to access registered OpenSSH nodes and will only happen when a user explicitly requests it (ie `tsh ssh -A`). The `forward_agent` role option will continue to be respected, allowing administrators to control if Teleport users can forward SSH agents.
```

Comment:
@capnspacehook Can you help clarify this:

> will only happen when a user explicitly requests it (ie `tsh ssh -A`)

What agent is forwarded in the `-A` case?  Forwarding the agent from the users computer is acceptable (since that would require them to go back to the proxy before connecting out).  But allowing `-A` to forward from the proxy certificate for the user would be concerning even if explicitly requested.

I assume that's not the case since you state `certificates signed by the OpenSSH CA will never be added to SSH agents`, but I would appreciate just clarifying that `-A` forwards only the source client agent like it does today.

Reply:
Only the user's agent will ever be forwarded, updated RFD to reflect this.